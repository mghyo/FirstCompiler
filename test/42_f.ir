#start_function main
void main():
int-list: 
float-list: $temp1, i 
  add, 40.0, 2.0, $temp1
  assign, i, $temp1, 
  call, printf, i
  return,,,
#end_function main
