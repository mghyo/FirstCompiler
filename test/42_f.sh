#!/bin/bash

set -e

./phase2 test/42_f.ir $1

diff out.s test/42_f.$1.s

spim -f out.s > tmp

diff tmp test/42_f.out

