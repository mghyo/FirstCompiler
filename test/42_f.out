Loaded: /usr/share/spim/exceptions.s
42.00000000